.github/workflows/sanitizer-build-and-test.yaml

```
@@ -0,0 +1,157 @@
+on:
+  schedule:
+    # run daily 20:00 on master branch
+    - cron: '0 20 * * *'
+  push:
+    branches:
+      - prerelease_test
+jobs:
+  sanitizer:
+    name: PG${{ matrix.pg }} ${{ matrix.name }} ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        name: ["Sanitizer"]
+        pg: ["12.7", "13.3"]
+        os: ["ubuntu-20.04"]
+        build_type: ["Debug"]
+
+        # llvm-symbolizer crashes so we don't use clang until we resolve that
+        extra_packages: ["clang-10 llvm-10 llvm-10-dev llvm-10-tools"]
+        llvm_config: [llvm-config-10"]
+        clang: ["clang-10"]
+
+        # gcc-10 is the minimum version that can parse the suppression files barely adequately
+        cc: ["gcc-10"]
+        cxx: ["g++-10"]
+
+    env:
+      PG_SRC_DIR: pgbuild
+      PG_INSTALL_DIR: postgresql
+      MAKE_JOBS: 6
+      CLANG: ${{ matrix.clang }}
+      CC: ${{ matrix.cc }}
+      CXX: ${{ matrix.cxx }}
+      # gcc CFLAGS, disable inlining for function name pattern matching to work for suppressions
+      CFLAGS: "-g -fsanitize=address,undefined -fno-omit-frame-pointer -O1 -fno-inline"
+      CXXFLAGS: "-g -fsanitize=address,undefined -fno-omit-frame-pointer -O1 -fno-inline"
+      # clang CFLAGS
+      #CFLAGS: "-g -fsanitize=address,undefined -fno-omit-frame-pointer -O1 -fno-inline-functions"
+      #CXXFLAGS: "-g -fsanitize=address,undefined -fno-omit-frame-pointer -O1 -fno-inline-functions"
+
+      # We do not link libasan dynamically to avoid problems with libdl and our libraries.
+      # clang does this by default, but we need to explicitly state that for gcc.
+      # static gcc LDFLAGS
+      LDFLAGS: "-fsanitize=address,undefined -static-libasan -static-liblsan -static-libubsan"
+      # static sanitizer clang LDFLAGS or dynamic sanitizer gcc LDFLAGS
+      #LDFLAGS: "-fsanitize=address,undefined"
+      ASAN_OPTIONS: suppressions=${{ github.workspace }}/scripts/suppressions/suppr_asan.txt detect_odr_violation=0 log_path=${{ github.workspace }}/sanitizer.log log_exe_name=true print_suppressions=false exitcode=27 external_symbolizer_path=/usr/lib/llvm-10/bin/llvm-symbolizer
+      LSAN_OPTIONS: suppressions=${{ github.workspace }}/scripts/suppressions/suppr_leak.txt print_suppressions=0 log_path=${{ github.workspace }}/sanitizer.log log_exe_name=true print_suppressions=false exitcode=27 external_symbolizer_path=/usr/lib/llvm-10/bin/llvm-symbolizer
+      UBSAN_OPTIONS: suppressions=${{ github.workspace }}/scripts/suppressions/suppr_ub.txt print_stacktrace=1 halt_on_error=1 log_path=${{ github.workspace }}/sanitizer.log log_exe_name=true print_suppressions=false exitcode=27 external_symbolizer_path=/usr/lib/llvm-10/bin/llvm-symbolizer
+
+    steps:
+    - name: Install Linux Dependencies
+      run: |
+        sudo apt-get update
+        sudo apt-get install flex bison lcov systemd-coredump gdb libipc-run-perl libtest-most-perl ${{ matrix.extra_packages }}
```

AFAICT, this will install different packages depending on the run, so *either* installing `clang-10`, ` llvm-10`, `llvm-10-dev`, or `llvm-10-tools`, but not all.

```suggestion
        sudo apt-get install flex bison lcov systemd-coredump gdb libipc-run-perl libtest-most-perl clang-10 llvm-10 llvm-10-dev llvm-10-tools
```